frontend/lib/resolution/Resolver.cpp

```diff
@@ -1592,25 +1592,23 @@ void Resolver::adjustTypesForSplitInit(ID id,
 
   // check to see if it is generic/unknown
   // (otherwise we do not need to infer anything)
-  auto g = Type::MAYBE_GENERIC;
-  if (lhsType.isUnknownKindOrType()) {
-     // includes nullptr type, UnknownType, and param with unknown value
-     g = Type::GENERIC;
-  } else {
-    CHPL_ASSERT(lhsType.type()); // should not be nullptr b/c of isUnknownKindOrType
-    g = getTypeGenericity(context, lhsType.type());
-  }
-
-  // return if there's nothing to do
-  if (g != Type::GENERIC) {
+  if (!(lhsType.isUnknownKindOrType() ||
+        getTypeGenericity(context, lhsType.type()) == Type::GENERIC))
```

Comment:
Sorry to be a nitpicker. IMO the DeMorgan'ed representation of this is equally understandable, and involves less parens.